main.tf

```diff
@@ -154,4 +155,14 @@ resource "kubernetes_secret" "secrets" {
   data = each.value
 }
 
+resource "google_compute_address" "ingress_external_ip" {
+  count  = var.create_static_ingress_ip ? 1 : 0
+  name   = local.ingress_ip_name
+  region = data.google_client_config.google_client.region
```

Comment:
Another point. We should add a `depends_on` in this resource on the node_pool to ensure we provision only if GKE provisioning is successful. What do you think?